topics/10_projects/main_building_a_project.md

```diff
@@ -1,3 +1,53 @@
 # Topic 10: Building a Test Automation Suite, utilizing both web and API testing
 
-Coming soon.
+This project is your chance to bring together everything you’ve learned during the training. You’ll gain hands-on experience by building a Test Automation Suite that includes both Web UI and API testing. The goal is to help you apply your skills in a real-world scenario and showcase your knowledge as a Test Automation Engineer.
+
+
+## Project Requirements
+
+The project should implement a Test Automation Framework Architecture where you can develop a suite of tests with the following components:
+
+
+### 1. Web UI Test Suite
+
+Create a suite of tests for a web user interface. At least two functionalities should be included. Feel free to use any testing framework you prefer.
+
+
+### 2. API Test Suite
+
+Develop a suite of tests for an API. This should include tests for at least these API methods: GET, POST, and PUT. You can also include tests for Data validation, and error handling. Use pytest and requests library for API testing.
+
+
+### 3. Documentation
+
+Provide basic documentation for your project:
+- Using docstring within your code
+- Creating a README.md file containing relevant information and instructions for running the project locally
+
+
+### 4. Code Quality
+
+Ensure your code quality by using tools like `pylint`/`flake8`/`black`/`pycodestyle`.
```

Comment:
One more suggestion - we will need to check the code. So let's define an explicit list of mandatory tools:
Instead of:
Ensure your code quality by using tools like `pylint`/`flake8`/`black`/`pycodestyle`.
Modify to:
The code of *.py files should produce no warnings or error while checking by:
1) pylint
2) flake8
3) black
4) pycodestyle

Reply:
Done